src/daft-scan/src/scan_task_iters/split_parquet_files_by_rowgroup.rs

```diff
@@ -0,0 +1,465 @@
+use std::{
+    collections::{HashMap, VecDeque},
+    sync::Arc,
+};
+
+use common_daft_config::DaftExecutionConfig;
+use common_error::DaftResult;
+use common_file_formats::{FileFormatConfig, ParquetSourceConfig};
+use common_runtime::get_io_runtime;
+use daft_io::{get_io_client, IOStatsContext};
+use daft_parquet::read::read_parquet_metadata;
+use daft_stats::TableMetadata;
+use futures::future::try_join_all;
+use itertools::Itertools;
+use parquet2::metadata::{FileMetaData, RowGroupList};
+
+use super::BoxScanTaskIter;
+use crate::{storage_config::StorageConfig, ChunkSpec, DataSource, ScanTask, ScanTaskRef};
+
+enum SplitSingleParquetFileByRowGroupsState {
+    Accumulating(usize, Vec<usize>, f64),
```

Comment:
would be nice to use struct syntax instead to know what each of these are

Reply:
also do you need a `Vec<usize>`, the range of row groups will always be contiguous right?

Shouldn't you be able to do `usize: start, usize: end` row